cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/environment/spark/SparkSubmitConfig.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.master.spi.environment.spark;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents environment specific spark submit configurations.
+ */
+public class SparkSubmitConfig {
+  private final String master;
+  private final Map<String, String> configs;
+
+  public SparkSubmitConfig(String master, Map<String, String> configs) {
+    this.configs = Collections.unmodifiableMap(new HashMap<>(configs));
+    this.master = master;
+  }
+
+  /**
+   * Returns spark master base path. This should be used to set spark master url:
+   * https://spark.apache.org/docs/latest/submitting-applications.html#master-urls
+   */
+  public String getMaster() {
+    return master;
+  }
+
+  /**
+   * Returns additional environment specific spark submit configurations. These will be added to --conf of spark submit.
```

Comment:
If it is always pass to spark-submit as `--conf`, shouldn't this class just be SparkConfig (similar to SparkConf, just that we don't want to bring in spark as dependency).

Reply:
Correct, thats what I called it originally. Though based on the method `getSparkSubmitConfig` name, I changed it to `SparkSubmitConfig`. Let me rename